src/user/tasks.py

```diff
@@ -0,0 +1,61 @@
+from celery.utils.log import get_task_logger
+
+from flask import current_app
+from flask_mail import Message
+
+from core import db, mail
+from core.logic import configure_mail_body
+from user.models import Role, User
+
+
+logger = get_task_logger(__name__)
+
+
+# @celery_app.task(name='approve-user')  # TODO (can't get celery tasks to work)
+def approve_user(user_id, awaiting_role_id, new_role_id, mail_context):
+    """ Sets user to 'approved' and emails them to let them know.
+
+    Args:
+        user_id (int): pk of user in database
+        awaiting_role_id (int): pk of the 'awaiting approval' Role
+        new_role_id (int): pk of the new Role to be assigned
+        mail_context (dict): variables to be used in the email templates
+    """
+
+    user = User.query.get(user_id)
+    user.roles.remove(
+        Role.query.get(awaiting_role_id).first()
+    )
+    user.roles.append(
+        Role.query.get(new_role_id).first()
+    )
+    user.approved = True
+    db.session.commit()
+
+    msg = Message(
+        "Altmetrics: Your account has been approved",
+        sender=current_app.config.get('MAIL_DEFAULT_SENDER'),
+        recipients=[user.email]
+    )
+    mail_context.update(user_name=user.first_name)
+    configure_mail_body(msg, 'account_approved', mail_context)
+
+    mail.send(msg)
+
+
+# @celery_app.task(name='send-approval-request')
+def send_approval_request():
+    """ Email site admin to let them know a new user has registered,
+    and their account requires approval.
+    """
+
+    msg = Message(
+        "Altmetrics: New user registered",
+        sender=current_app.config.get('MAIL_DEFAULT_SENDER'),
+        # recipients=[current_app.config.get('TECH_EMAIL')]
+        recipients=['g07h3212@gmail.com']
```

Comment:
Am I right in assuming that `recipients` will eventually be taken from config?

Reply:
Yes, sorry this was updated in the last commit :+1: